src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift

```diff
@@ -950,5 +1110,33 @@ private extension URLRequest {
 		guard let data = (String.getRandomString(of: 28 * paddedKeysAmount)).data(using: .ascii) else { return Data() }
 		return data
 	}
+
+	// MARK: - Multipart form-data
+
+	/// Constructs a `multipart/form-data` body for a given log data
+	/// - Parameters:
+	///   - logData: The compressed log data to submit
+	///   - boundary: Delimiter in multipart documents; see: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#multipartform-data
+	/// - Throws: A `DataConversionError` if given strings cannot be decoded to UTF-8 data
+	/// - Returns: The prepared body data for the http request
+	private static func requestBodyForLogUpload(
+		logData: Data,
+		boundary: String
+	) throws -> Data? {
+		var body = Data()
+
+		// init form
+		try body.append("\r\n--\(boundary)\r\n")
+		// prevent potential file collisions on backend
+		try body.append("Content-Disposition:form-data; name=\"file\"; filename=\"ErrorLog-\(UUID().uuidString).zip\"\r\n")
+		try body.append("Content-Type:application/zip\r\n")
+		try body.append("Content-Length: \(logData.count)\r\n")
+		try body.append("\r\n")
+		body.append(logData)
+		try body.append("--\(boundary)--\r\n")
```

Comment:
```suggestion
		try body.append("\r\n")
		try body.append("--\(boundary)--\r\n")
```

Reply:
Fails against the mock server without that change

Reply:
Indeed, fixes the 500 error.